code/modules/antagonists/voidwalker/voidwalker_mob.dm

```diff
@@ -0,0 +1,336 @@
+	new /obj/effect/temp_visual/circle_wave/unsettle(get_turf(victim))
+
+	if(!SSmapping.lazy_load_template(LAZY_TEMPLATE_KEY_VOIDWALKER_VOID) || !GLOB.voidwalker_void.len)
+		victim.forceMove(get_random_station_turf())
+		victim.heal_overall_damage(brute = 80, burn = 20)
+		CRASH("[victim] was instantly dumped after being voidwalker kidnapped due to a missing landmark!")
+	else
+		victim.heal_and_revive(90)
+		victim.adjustOxyLoss(-100, FALSE)
+
+		conversions_remaining++
+
+		var/obj/wisp = new /obj/effect/wisp_mobile (get_turf(pick(GLOB.voidwalker_void)))
+		victim.forceMove(wisp)
+
+/// Check if theyre still incapacitated for the kidnap do_after
+/mob/living/basic/voidwalker/proc/check_incapacitated(mob/living/carbon/human/kidnappee)
+	return kidnappee.incapacitated
+
+/// Modding the voidwalker is funny, so setting the home_turf sets everything right for easy of modding
+/mob/living/basic/voidwalker/vv_edit_var(vname, vval)
+	. = ..()
+	// This is all very snowflakey code but like, it's supposed to be. It's just for helping admins mod it
+	if(vname == NAMEOF(src, home_turf))
+		var/datum/component/space_camo/camo = GetComponent(/datum/component/space_camo)
+		var/datum/component/space_dive/dive = GetComponent(/datum/component/space_dive)
+
+		camo.camo_tile = vval
+		dive.diveable_turf = vval
+		if(istype(charge, /datum/action/cooldown/mob_cooldown/charge/voidwalker))
+			var/datum/action/cooldown/mob_cooldown/charge/voidwalker/charge_voidwalker = charge
+			charge_voidwalker.valid_target_turf = vval
+
+		if(!isspaceturf(vval))
+			qdel(GetComponent(/datum/component/planet_allergy))
+			remove_movespeed_modifier(speed_modifier)
+			speed_modifier = null
+
+///
+/// Wall Conversion
+///
+
+/// Attempt to convert a wall into passable voidwalker windows
+/mob/living/basic/voidwalker/proc/try_convert_wall(turf/closed/wall/our_wall)
+	if(!conversions_remaining)
+		balloon_alert(src, "need more kidnaps!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!COOLDOWN_FINISHED(src, wall_conversion))
+		balloon_alert(src, "must wait [DisplayTimeText(COOLDOWN_TIMELEFT(src, wall_conversion))]!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!check_wall_validity(our_wall, src, silent = FALSE))
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+	playsound(our_wall, 'sound/effects/magic/blind.ogg', 100, TRUE)
+	new /obj/effect/temp_visual/transmute_tile_flash(our_wall)
+	balloon_alert(src, "opening window...")
+	if(!do_after(src, 8 SECONDS, our_wall, hidden = TRUE))
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+	if(!conversions_remaining)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	var/list/target_walls = list()
+	target_walls += our_wall
+	for(var/turf/closed/wall/adjacent_wall in orange(1, our_wall))
+		if(check_wall_validity(adjacent_wall))
+			target_walls += adjacent_wall
+
+	for(var/turf/closed/wall/targeted_wall in target_walls)
+		playsound(targeted_wall, 'sound/effects/magic/blind.ogg', 100, TRUE)
```

Comment:
Do we want to play the sound from every wall? given that it's at most a 3x3 square with no delay in playing the sounds that will result in a single very loud sound.